services/src/main/java/edu/unc/lib/dl/cdr/services/processing/SetPrimaryObjectService.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2008 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.cdr.services.processing;
+
+import static edu.unc.lib.dl.acl.util.Permission.editResourceType;
+
+import edu.unc.lib.dl.acl.service.AccessControlService;
+import edu.unc.lib.dl.acl.util.AgentPrincipals;
+import edu.unc.lib.dl.fcrepo4.FileObject;
+import edu.unc.lib.dl.fcrepo4.RepositoryObject;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectFactory;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectLoader;
+import edu.unc.lib.dl.fcrepo4.WorkObject;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.model.InvalidOperationForObjectType;
+import edu.unc.lib.dl.rdf.Cdr;
+import edu.unc.lib.dl.rdf.Premis;
+
+/**
+ * Service that manages setting a primary object on a work
+ *
+ * @author harring
+ *
+ */
+public class SetPrimaryObjectService {
+
+    private AccessControlService aclService;
+    private RepositoryObjectLoader repoObjLoader;
+    private RepositoryObjectFactory repoObjFactory;
+
+    /**
+     * Sets file object with the given pid as the primary object for its work using the agent principals provided.
+     *
+     * @param agent security principals of the agent making request.
+     * @param fileObjPid the file object to be set as primary object
+     */
+    public void setPrimaryObject(AgentPrincipals agent, PID fileObjPid) {
+
+        aclService.assertHasAccess("Insufficient privileges to set " + fileObjPid.getUUID() + " as primary object",
+                fileObjPid, agent.getPrincipals(), editResourceType);
+
+        RepositoryObject repoObj = repoObjLoader.getRepositoryObject(fileObjPid);
+
+        if (!(repoObj instanceof FileObject)) {
```

Comment:
It would actually make sense for work.setPrimaryObject to enforce this restriction

Reply:
done!